benchmarks/distributed/dataloading.py

```diff
@@ -53,38 +43,31 @@
     type=int,
     default=2,
 )
-
+parser.add_argument(
+    "--wandb_entity",
+    type=str,
+    default="",
+)
+parser.add_argument(
+    "--wandb_key",
+    type=str,
+    default="",
+)
 torch.cuda.set_device(0)
 
 RETRY_LIMIT = 2
-NUM_WORKERS = 16
+NUM_WORKERS = 64
```

Comment:
These cores are used only to build/preprocess the dataset "once and for all" using plain torch dataloader (ie you would do that once, and hopefully only once per paper :p ). We include only the deterministic transforms, and leave the random ones for the "real" training loop.
Then you have a contiguous mem representation of your data on disk that you can play with.

Later (during the "real" training loop), you use only as many procs as your world size.

We use 64 because we run it often and it's slightly faster than 16 or 32, albeit not significantly. 

Now that being said we could lower this number even to 1 or 0, it would just make running the benchmark longer but the later stage would still be super fast. FYI we get to 30K fps with this (>10x faster than traditional transforms).
Does that make sense?

With @tcbegley we wanted to add a comment about the "cost" of using gpus for preproc: here we have 1GPU / subcollector. I'm a bit surprised by how much mem each proc consumes, but even there: if your training loop with imagenet is dataloading-bound, you could speed it up by -- say -- 5x using what we present here, thereby reducing the overall cost. Intuitively, even if your exp costs more, you run it for a shorter time (because you use your resources better), and you also gain in dev/ops time.

Reply:
Thanks for the clarification